target/i386/tcg/translate.c

```diff
@@ -304,6 +305,82 @@ static const uint8_t cc_op_live[CC_OP_NB] = {
     [CC_OP_POPCNT] = USES_CC_SRC,
 };
 
+
+#ifdef HAS_TRACEWRAP
+#include <frame_arch.h>
+
+static void gen_trace_newframe(DisasContext *ctx) {
+    TCGv_i64 pc_tcg = tcg_temp_new_i64();
+    tcg_gen_movi_i64(pc_tcg, ctx->pc);
```

Comment:
```suggestion
    tcg_gen_movi_i64(pc_tcg, ctx->base.pc_next);
```
This will fix incorrect addresses in the frames.